core/src/main/java/org/apache/druid/java/util/common/Bytes.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.common;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import java.io.Serializable;
+import java.util.Locale;
+
+@JsonSerialize(using = BytesSerializer.class)
+public class Bytes implements Serializable
+{
+  public static final Bytes ZERO = new Bytes(0L);
+
+  private long value;
+
+  public Bytes(String value)
+  {
+    this.value = Bytes.parse(value);
+  }
+
+  public Bytes(long value)
+  {
+    this.value = value;
+  }
+
+  public long getValue()
+  {
+    return value;
+  }
+
+  @Override
+  public boolean equals(Object thatObj)
+  {
+    if (thatObj == null) {
+      return false;
+    }
+    if (thatObj instanceof Bytes) {
+      return value == ((Bytes) thatObj).value;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Long.hashCode(value);
+  }
+
+  @Override
+  public String toString()
+  {
+    return String.valueOf(value);
+  }
+
+  public static Bytes valueOf(int bytes)
+  {
+    return new Bytes(bytes);
+  }
+
+  public static Bytes valueOf(long bytes)
+  {
+    return new Bytes(bytes);
+  }
+
+  public static long parse(String number)
+  {
+    if (number == null) {
+      throw new IAE("number is null");
```

Comment:
All error messages should start with a capital letter. Please fix other places too.